added updated nft guide (#74)
nftlabs/portal: portal/guides/nft-gated-website.mdx

---

> title: How to Create An NFT-Gated Website

Reviewer: ```suggestion
title: How to create an NFT-gated website
```
We don't use TitleCas

author: "Samina and Nacho Iacovino"
authorAvatar: "/assets/portal/authors/samina.jpeg"
authorAvatar2: "/assets/portal/authors/nacho-iacovino.jpg"
authorTwitter: "https://twitter.com/saminacodes"
authorTwitter2: "https://twitter.com/nachoiacovino"
draft: false

> tags: [NFTs, mint, access]

Reviewer: ```suggestion
tags: [nft-drop, gated, mint, access]
```

summary: "In this guide, we will create a website that restricts content based on owning an NFT using React."
image: "/assets/portal/guides/nft-gated-website/thumbnail.png"
date: "2021-09-16"
updated: "2022-04-12"
---

One of the more dynamic use cases for NFTs is using them as a membership pass to the NFT holders. Let’s assume you want to create a website for your community that is gated by having access to a specific NFT from a collection. You can keep downloadable content or hidden pages by access to the NFT.

In this guide, we will create a website that restricts content based on owning an NFT using React.

> You can grab a copy of the final project from our examples repository: [https://github.com/thirdweb-example/NFT-Gated-Website](https://github.com/thirdweb-example/NFT-Gated-Website).

Reviewer: ```suggestion
You can grab a copy of the final project from our examples organization, [check it out here](https://github.com/thirdweb-example/NFT-Gated-Website).
```

<MdxAlert status="warning">
If your NFTs are an NFT Drop or Edition Drop contract, set up claim phases before using the claiming functions mentioned later in the guide.
</MdxAlert>

For our example, I am using an NFT Drop contract called Cookie Club for members of the Cookie Club. 🤫

Link to NFT Drop dashboard for Cookie Club: [https://thirdweb.com/dashboard/rinkeby/nft-drop/0x66463b3C1EBf08b9dE889BCc0A5cbf29dc0e2B7a?tabIndex=0](https://thirdweb.com/dashboard/rinkeby/nft-drop/0x66463b3C1EBf08b9dE889BCc0A5cbf29dc0e2B7a?tabIndex=0)

## Clone the template repository

To get started will use a `cra-javascript-template` which already includes a working SDK setup. This template will make it easy to learn and save time when getting started with our project.

> <MdxAlert status="info">

Reviewer: if you mention this you can also add a snippet of index.js, it's short and it will give all the tools for people to run it themselves

Author: I am trying to push them to clone the repository in the paragraph

Reviewer: I know, but it's very easy to add index.js here, it's just like adding 10 lines and gives more value.

Template is the easier way, but people are free to not use it

If you’d prefer to use your own template, you can install the latest version of our SDK in your project with `npm install @thirdweb-dev/sdk react ethers` or`yarn add @thirdweb-dev/sdk react ethers`

</MdxAlert>

> First, navigate to [https://github.com/thirdweb-example/cra-javascript-starter](https://github.com/thirdweb-example/cra-javascript-starter) and click on "Use this template" to create a copy.

Reviewer: ```suggestion
First, navigate to [this GitHub repository](https://github.com/thirdweb-example/cra-javascript-starter) and click on "Use this template" to create a copy.
```

![thirdweb Create React App starter template on GitHub](/assets/portal/guides/nft-gated-website/cra-javascript-starter.png)

Add the project’s name, description, and other settings, and then clone it locally on your device. Open the directory and run `npm install` in your terminal to install all the dependencies.

## Setup the blockchain

Inside our `src` folder, we will open the `index.js` file. We will first change the chain id to the chain of our NFT Drop. For this project, we’ll be using `Rinkeby` since our NFT Drop contract is on the Rinkeby testnet.

> ```markdown

Reviewer: ```suggestion
```jsx
```

> const activeChainId = ChainId.Rinkeby;

Reviewer: replace this with index.js then

```

## Setup the connect wallet button

In the same folder, there is the `app.js` file. In this one, we want to allow our user to connect to the app and then obtain our user’s wallet address. On the front end, we will create a button that lets users connect with MetaMask. When a wallet is connected, it will display the corresponding address.

> ```jsx

Reviewer: ```suggestion
```jsx:App.js
```

Add file name

import { useAddress, useMetamask } from "@thirdweb-dev/react";
import "./styles.css";

const App = () => {
  // get address of user & allow them to connect with metamask
  const address = useAddress();
  const connectWithMetamask = useMetamask();

  //if there isn't a wallet connected, display our connect MetaMask button
  if (!address) {
    return (
      <>
        <h1>Welcome to the Cookie Club</h1>
        <button className="btn" onClick={connectWithMetamask}>
          Connect MetaMask
        </button>
      </>
    );
  }

  // if an address is connected, display address
  return (
    <div>
      <p>Your address: {address}</p>
    </div>
  );
};

export default App;
```

Let’s preview what our app looks like and ensure it’s working by running `npm start` in the terminal.

![Welcome screen for application with Connect Wallet button](/assets/portal/guides/nft-gated-website/welcome-screen.png)

> ## Conditionally Render Content & Add an NFT Minting button

Reviewer: Don't use TitleCase here

In the same `App.js` file, we will add conditionals that render specific pages based on the connection status of the user. We want the user to be able to mint an NFT from the drop if they don’t already hold an NFT. If the connected user does have one, we will display some congratulatory text.
As an extra, I added a helper function to display our shortened wallet address called `truncateAddress`. Feel free to add or omit this as well in your app.

> ```jsx

Reviewer: ```suggestion
```jsx:App.js
```

  }

	// if the user is connected and has an NFT from the drop, display text
	if (hasClaimedNFT) {
    return <h2>Congratulations! You have a Cookie NFT! 🍪</h2>;
  }

  // helper function to truncate the address so it displays in a nice format
  function truncateAddress(address) {
    return `${address.slice(0, 6)}...${address.slice(-4)}`;
  }

	//if the user does not have an NFT, show their address and mint an NFT button
	return (
    <>
      <p className="address">
        There are no Cookie NFTs held by:{" "}
        <span className="value">{truncateAddress(address)}</span>
      </p>
      <button className="btn mint" disabled={isClaiming} onClick={mintNft}>
        Mint NFT
      </button>
    </>
  );
}

export default App;
```

![No NFTs are currently owned screen](/assets/portal/guides/nft-gated-website/none-held-screen.png)

## Add useEffect to check for state changes

Lastly, we want to add a `useEffect` function to ensure we are up to date with the latest state of the app. This function triggers every time something in the dependency array changes. For example, if the user’s address or `nftDrop` disconnects or changes, we want to refresh and update it accordingly.

> ```jsx

Reviewer: ```suggestion
```jsx:App.js
```
